Sources/Quick/QuickSpec.m

```diff
@@ -115,10 +115,23 @@ + (SEL)addInstanceMethodForExample:(Example *)example {
     }
 
     const char *types = [[NSString stringWithFormat:@"%s%s%s", @encode(id), @encode(id), @encode(SEL)] UTF8String];
-    NSString *selectorName = [NSString stringWithFormat:@"%@_%@_%ld",
-                              example.name.qck_selectorName,
-                              sanitizedFileName,
-                              (long)example.callsite.line];
+    
+    NSString *originalName = example.name.qck_selectorName;
+    NSString *selectorName = originalName;
+    unsigned int i = 2;
+    
+    static NSMutableSet<NSString *> *selectorNames;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        selectorNames = [[NSMutableSet alloc] init];
+    });
+    
+    while ([selectorNames containsObject:selectorName]) {
+        selectorName = [NSString stringWithFormat:@"%@_%u", originalName, i++];
+    }
```

Comment:
Thanks for the bump, I indeed hadn't noticed. You may be able to test the selector name using example metadata, as in this example: https://github.com/Quick/Quick/blob/662950f3e14329ce7569f940e4cce17b93928e4a/Sources/QuickTests/FunctionalTests/ItTests.swift#L17-L20

If that doesn't work, perhaps we should think about testing the test output, as in https://github.com/apple/swift-corelibs-xctest/pull/20. It doesn't have to be in this pull request, though--if we can't test the selector in an automated way, so be it.